design/repo_maintenance_job_config.md

```diff
@@ -0,0 +1,144 @@
+	MemLimit            string `json:"memLimit,omitempty"`
+}
+
+type LoadAffinity struct {
+    // NodeSelector specifies the label selector to match nodes
+    NodeSelector metav1.LabelSelector `json:"nodeSelector"`
+}
+```
+
+The `LoadAffinity` structure is reused from design [node-agent affinity configuration](Implemented/node-agent-affinity.md).
+Users may want to set different LoadAffinity configurations according to different conditions (i.e., for different storages represented by StorageClass, CSI driver, etc.), so we define ```loadAffinity``` as an array.
+This is for extensibility consideration, at present, we don't implement multiple configurations support, so if there are multiple configurations, we always take the first one in the array.
+When implementing this design, some code will be refactored as utility functions to make them reusable.
+
+### Sample
+A sample of the ```repo-maintenance-job-config``` configMap is as below:
+``` bash
+cat <<EOF > repo-maintenance-job-config.json
+{
+    "keepLatestJobsCount": 2,
+    "cpuRequest": "100m",
+    "cpuLimit": "200m",
+    "memRequest": "100Gi",
+    "memLimit": "200Gi",    
+    "loadAffinity": [
+        {
+            "nodeSelector": {
+                "matchLabels": {
+                    "beta.kubernetes.io/instance-type": "Standard_B4ms"
+                },
+                "matchExpressions": [
+                    {
+                        "key": "kubernetes.io/hostname",
+                        "values": [
+                            "node-1",
+                            "node-2",
+                            "node-3"
+                        ],
+                        "operator": "In"
+                    },
+                    {
+                        "key": "xxx/critial-workload",
+                        "operator": "DoesNotExist"
+                    }
+                ]          
+            }
+        }
+    ]
+}
+EOF
+```
+This sample showcases two affinity configurations:
+- matchLabels: VGDP instances will run only in nodes with label key `beta.kubernetes.io/instance-type` and value `Standard_B4ms`
+- matchExpressions: VGDP instances will run in node `node1`, `node2` and `node3` (selected by `kubernetes.io/hostname` label)
+
+This sample showcases one anti-affinity configuration:
+- matchExpressions: VGDP instances will not run in nodes with label key `xxx/critial-workload`
+
+To create the configMap, users need to save something like the above sample to a json file and then run below command:
+```
+kubectl create cm repo-maintenance-job-config -n velero --from-file=repo-maintenance-job-config.json
+kubectl -n velero label cm repo-maintenance-job-config velero.io/config-map-type=repo-maintenance-job-config
+``` 
+
+### Implementation
+During the Velero server start,  `velero server` CLI uses this function calling flow to initializes the `Repository.Manager`  `run -> s.initRepoManager -> repository.NewManager`.
```

Comment:
> During the Velero server start

As mentioned above, I don't think this is necessary. However, it is somehow awkward for `KeepLatestJobsCount`, if we load the configurations dynamically, we get a dynamic number of `KeepLatestJobsCount`, so it may make the decision logic complex.

Furthermore, we may be able to remove the need to keep some completed jobs in future, so I suggest we don't add this `KeepLatestJobsCount` into the configuration.